src/sdk/v2/permit.ts

```diff
@@ -387,20 +387,21 @@ export class PermitV2 implements PermitV2Interface, PermitV2Metadata {
   unseal<T extends any[]>(item: [...T]): [...MappedUnsealedTypes<T>];
   unseal<T>(item: T) {
     // SealedItem
-    if (isSealedItem(item)) {
+    const sealedItem = getAsSealedItem(item);
+    if (sealedItem != null) {
       const bn = chainIsHardhat(this._signedChainId)
-        ? hardhatMockUnseal(item.data)
-        : this.sealingPair.unseal(item.data);
+        ? hardhatMockUnseal(sealedItem.data)
+        : this.sealingPair.unseal(sealedItem.data);
 
-      if (isSealedBool(item)) {
+      if (isSealedBool(sealedItem)) {
         // Return a boolean for SealedBool
         return Boolean(bn).valueOf() as any;
       }
-      if (isSealedAddress(item)) {
+      if (isSealedAddress(sealedItem)) {
         // Return a string for SealedAddress
         return getAddress(`0x${bn.toString(16).slice(-40)}`) as any;
       }
-      if (isSealedUint(item)) {
+      if (isSealedUint(sealedItem)) {
```

Comment:
This section updates how the `SealedUint/Bool/Address` return types are matched and unsealed. This ensures that the return type from `ethers` contract calls can be recognized as a `SealedUint/Bool/Address` during automatic unsealing.